fe/fe-core/src/main/java/com/starrocks/connector/MvPartitionConverter.java

```diff
@@ -0,0 +1,176 @@
+        }
+
+        private static PartitionKey convertToDate(PartitionKey partitionKey) {
+            PartitionKey newPartitionKey = new PartitionKey();
+            String dateLiteral = partitionKey.getKeys().get(0).getStringValue();
+            LocalDateTime dateValue = DateUtils.parseStrictDateTime(dateLiteral);
+            try {
+                newPartitionKey.pushColumn(new DateLiteral(dateValue, Type.DATE), PrimitiveType.DATE);
+                return newPartitionKey;
+            } catch (AnalysisException e) {
+                throw new SemanticException("create date string:{} failed:",
+                        partitionKey.getKeys().get(0).getStringValue(),
+                        e);
+            }
+        }
+
+        @Override
+        public Map<String, Range<PartitionKey>> convert(Map<String, PartitionKey> sourcePartitions) {
+            Map<String, Range<PartitionKey>> mvPartitionRangeMap = new LinkedHashMap<>();
+            PrimitiveType resultColumnType = isConvertToDate ? PrimitiveType.DATE : partitionColumnType;
+
+            // sort the partitions by value
+            LinkedHashMap<String, PartitionKey> sortedPartitionLinkMap = sourcePartitions.entrySet().stream()
+                    .sorted(Map.Entry.comparingByValue(PartitionKey::compareTo))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1,
+                            LinkedHashMap::new));
+
+            // build the range map by discrete values
+            int index = 0;
+            PartitionKey lastPartitionKey = null;
+            String lastPartitionName = null;
+            for (Map.Entry<String, PartitionKey> entry : sortedPartitionLinkMap.entrySet()) {
+                if (index == 0) {
+                    lastPartitionName = entry.getKey();
+                    lastPartitionKey = isConvertToDate ? convertToDate(entry.getValue()) : entry.getValue();
+                    if (lastPartitionKey.getKeys().get(0).isNullable()) {
+                        // If partition key is NULL literal, rewrite it to min value.
+                        lastPartitionKey = PartitionKey.createMinPartition(ImmutableList.of(resultColumnType));
+                    }
+                    ++index;
+                    continue;
+                }
+                Preconditions.checkState(!mvPartitionRangeMap.containsKey(lastPartitionName));
+                PartitionKey upperBound = isConvertToDate ? convertToDate(entry.getValue()) : entry.getValue();
+                mvPartitionRangeMap.put(lastPartitionName, Range.closedOpen(lastPartitionKey, upperBound));
+                lastPartitionName = entry.getKey();
+                lastPartitionKey = upperBound;
+            }
+            if (lastPartitionName != null) {
+                PartitionKey endKey = new PartitionKey();
+                endKey.pushColumn(PartitionUtil.addOffsetForLiteralUnchecked(lastPartitionKey.getKeys().get(0), 1),
```

Comment:
if one partition for each two days is created automatically,  
current latest partition range [lastPartitionKey, lastPartitionKey+1)  would leave a hole [lastPartitionKey+1, lastPartitionKey+2); is it acceptable?